GDAL-3.9.1 compilation error

### What is the bug?

GDAL-3.9.1 compilation error :
```
gdal-3.9.1/port/cpl_vsil_libarchive.cpp: Dans la fonction « archive* VSICreateArchiveHandle(const string&) »:
gdal-3.9.1/port/cpl_vsil_libarchive.cpp:191:9: error: « archive_read_support_format_rar5 » n'a pas été déclaré dans cette portée
         archive_read_support_format_rar5(pArchive);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gdal-3.9.1/port/cpl_vsil_libarchive.cpp:191:9: note: alternatives suggérées: « archive_read_support_format_rar »
         archive_read_support_format_rar5(pArchive);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```


### Steps to reproduce the issue

```
> cat /etc/redhat-release
Red Hat Enterprise Linux release 8.10 (Ootpa)
```

`> cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local -DTIFF_INCLUDE_DIR=/usr/local/include -DTIFF_LIBRARY_RELEASE=/usr/local/lib/libtiff.so`

### Versions and provenance

Version 3.9.1 from : `http://download.osgeo.org/gdal/3.9.1/gdal-3.9.1.tar.gz` (idem with 3.9.0)

### Additional context

_No response_

--- a/port/cpl_vsil_libarchive.cpp
+++ b/port/cpl_vsil_libarchive.cpp
@@ -188,7 +188,9 @@ static struct archive *VSICreateArchiveHandle(const std::string &osFSPrefix)
     else
     {
         archive_read_support_format_rar(pArchive);
+#ifdef ARCHIVE_FORMAT_RAR_V5
         archive_read_support_format_rar5(pArchive);
+#endif
     }
 
     return pArchive;

